api/applications.go

```diff
@@ -210,12 +178,38 @@ func (c *MSGraphClient) RemoveApplicationPassword(ctx context.Context, applicati
 func getPasswordCredentialsForApplication(app models.Applicationable) []PasswordCredential {
 	var appCredentials []PasswordCredential
 	for _, cred := range app.GetPasswordCredentials() {
-		appCredentials = append(appCredentials, PasswordCredential{
-			SecretText:  *cred.GetSecretText(),
-			EndDate:     *cred.GetEndDateTime(),
-			KeyID:       cred.GetKeyId().String(),
-			DisplayName: *cred.GetDisplayName(),
-		})
+		appCredentials = append(appCredentials, getPasswordCredentialResponse(cred))
 	}
 	return appCredentials
 }
+
+func getApplicationResponse(app models.Applicationable) Application {
+	if app == nil {
+		return Application{
+			AppID:               "",
+			AppObjectID:         "",
+			PasswordCredentials: []PasswordCredential{},
+		}
+
+	}
+	return Application{
+		AppID:               *app.GetAppId(),
+		AppObjectID:         *app.GetId(),
```

Comment:
In theory it is still possible to get a panic if GetId or GetAppId return nil. In my case, the panic I got was when `app` as non-nil but GetDescription returned nil. So we should probably make sure these function calls are non-nil before we dereference them just in case. :)

Reply:
Ah interesting, good to know. I added some extra guardrails on the function calls themselves to prevent any nil pointer dereferencing :)